src/services/TechRecordsService.ts

```diff
@@ -185,6 +178,48 @@ class TechRecordsService {
         throw new HTTPError(500, HTTPRESPONSE.INTERNAL_SERVER_ERROR);
       });
   }
+
+  public updateVin(vehicle: Vehicle, newVin: string) {
+    const vehicleClone = cloneDeep(vehicle);
+
+    const oldVehicle: Vehicle = { ...vehicleClone, techRecord: [] };
+    const newVehicle: Vehicle = {
+      ...vehicleClone,
+      vin: newVin,
+      techRecord: [],
+    };
+    let current: TechRecord | undefined;
+    let provisional: TechRecord | undefined;
+
+    vehicleClone.techRecord.forEach((record) => {
+      switch (record.statusCode) {
+        case STATUS.PROVISIONAL:
+          provisional = { ...record };
+          break;
+        case STATUS.CURRENT:
+          current = { ...record };
+          break;
+        case STATUS.ARCHIVED:
+          oldVehicle.techRecord.push({ ...record });
+          break;
+        default:
+          break;
+      }
+    });
+
+    if (current && provisional) {
+      newVehicle.techRecord.push({ ...current });
+      oldVehicle.techRecord.push({ ...current, statusCode: "archived" });
+    } else if (provisional) {
+      newVehicle.techRecord.push({ ...provisional });
+      oldVehicle.techRecord.push({ ...provisional, statusCode: "archived" });
+    } else if (current) {
+      newVehicle.techRecord.push({ ...current });
+      oldVehicle.techRecord.push({ ...current, statusCode: "archived" });
+    }
+
+    return [oldVehicle, newVehicle];
```

Comment:
```suggestion
    return {oldVehicle, newVehicle};
```
I think it would be nicer to return an object here. It can still be destructured in the calling function (as it is now) however, it is nicer to do:
```js
const oldVehicle = updatedVins.oldVehicle
```
over
```js
const oldVehicle = updatedVins[0]
```
although I understand we might not need any of this so yeah kind of a nitpick